server/routes/timetableItemBuilder.ts

```diff
@@ -0,0 +1,184 @@
+import { format, parseISO } from 'date-fns'
+import { SessionSchedule } from '../data/orchestrationApiTypes'
+import { prisonerTimePretty } from '../utils/utils'
+
+export type TimetableItem = {
+  time: string
+  type: 'Open' | 'Closed'
+  capacity: string
+  attendees: string
+  frequency: string
+  endDate: string
+}
+// Builds timetable rows, using all session schedules for the selected date
+export default ({
+  schedules,
+  selectedDate,
+}: {
+  schedules: SessionSchedule[]
+  selectedDate: string
+}): TimetableItem[] => {
+  const dateFormat = 'd MMMM yyyy'
+  const timetableItems: TimetableItem[] = []
+  schedules.forEach(schedule => {
+    const time = `${prisonerTimePretty(`${selectedDate}T${schedule.sessionTimeSlot.startTime}`)} to ${prisonerTimePretty(`${selectedDate}T${schedule.sessionTimeSlot.endTime}`)}`
+
+    const endDate = schedule.sessionDateRange.validToDate
+      ? format(parseISO(schedule.sessionDateRange.validToDate), dateFormat)
+      : 'Not entered'
+
+    let frequency = ''
+    if (
+      schedule.sessionDateRange.validToDate &&
+      schedule.sessionDateRange.validFromDate === schedule.sessionDateRange.validToDate
+    ) {
+      frequency = 'One off'
+    } else {
```

Comment:
Can always set `frequency` to this `else` value and ditch the `else`. Tidies it up a bit.